cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,45 @@
-/// <reference types='cypress' />
-/// <reference types='../support' />
-
-describe('Sign Up page', () => {
-  before(() => {
-
-  });
-
-  it('should ...', () => {
-
-  });
-});
+/// <reference types='cypress' />
+/// <reference types='../support' />
+
+import SignUpPageObject from "../support/pages/signUp.PageObject";
+import HomePageObject from "../support/pages/home.pageObject";
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
+describe("Sign Up page", () => {
+  let user;
+  let newUser;
+  beforeEach(() => {
+    cy.task("db:clear");
+    cy.task("generateUser").then((generatedUser) => {
+      user = generatedUser;
+    });
+    cy.task("generateUser").then((generatedUser) => {
+      newUser = generatedUser;
+    });
+    signUpPage.visit();
+  });
+
+  it("should provide an ability to sign up with valid credentials", () => {
+    signUpPage.visit();
+    cy.getByDataCy("signup-username-field").should("be.visible");
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.assertSuccessfulMessage();

```

Comment:
Ensure that the `assertSuccessfulMessage` function is correctly implemented to verify that the appropriate success message is displayed when sign-up is successful.